bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BUnionType.java

```diff
@@ -307,7 +313,12 @@ public <V extends Object> V getEmptyValue() {
             return null;
         }
 
-        return memberTypes.get(0).getEmptyValue();
+        if (memberTypes.get(0).getTag() == TypeTags.FINITE_TYPE_TAG) {
```

Comment:
Seems this if condition does not covered by tests. Do we need this?

Reply:
This whole `getEmptyValue()` method has not been covered by tests previously also. It seems this method is not getting called during code execution, only `getZeroValue()` method. Looking at the way the code has been written previously, I think the changes have to be added here also, the difference being, this will call `getEmptyValue()` again and the other will call `getZeroValue()`.